app/controllers/videos_controller.rb

```diff
@@ -19,4 +19,20 @@ def toggle_star
     @video.toggle :starred
     @video.save
   end
+
+  #handle file uploads
+  def import   
+    if request.post?
+      file_extension = File.extname(params[:file].original_filename)
+        if file_extension == ".csv"
+          if Video.import(params[:file]) != false
+            redirect_to videos_path, notice: "Video(s) imported!"
+          else
+            redirect_to import_videos_path, alert: "Need youtube_id column/value in CSV file"
```

Comment:
Much of the logic should lie in the model. This means raising exceptions and checking file extensions should go there.